lib/Memecoin.ts

```diff
@@ -466,39 +481,67 @@ export class MemecoinSDK {
     return BigInt(result)
   }
 
-  private async estimateSwapCoin(params: EstimateSwapCoinParams): Promise<bigint> {
-    const { fromToken, toToken, amountIn } = params
+  async estimateSwap(params: EstimateSwapParams): Promise<EstimateSwapResult> {
```

Comment:
create a class called `TokenSwapper`, create an instance of it in the constructor. Ultimately we should see 

```typescript
return this.swapper.estimateSwap(params)
```